### DISCUSSION TITLE
How Install CGAL 6.0 in windows

### DISCUSSION THREAD

**Discussion by DarkSun812** [CGAL/cgal#8538] [General]
I am installing CGAL 6.0 for windows and running a sample code that is in the official documentation, but it does not run as I get the following error 

***===== Error 1 =====***
```
Severity Code Description Project Project File Line Status suppressed Details
Error LNK2019 unresolved external symbol __imp___gmpz_clear referenced in function “public: __cdecl boost::multiprecision::backends::gmp_int::~gmp_int(void)” (??1gmp_int@backends@multiprecision@boost@@QEAA@XZ) CGAL TEST C:\Users\baezaezaezaezaeource\reposseCGAL TEST\CGAL TEST.obj 1	
```	

***==============*

Steps I used to install CGAL 6.0:

- I am using visual studio 2022, I downloaded the source “CGAL-6.0-library.zip” and “GMP and MPFR libraries, for Windows 64bits ‘which is available at the following official link: ’https://github.com/CGAL/cgal/releases” (Note that there is only one gmp folder, I didn't see any that said MPFR). 
- grab the “auxiliary” folder from the gmp file and paste it inside the CGAL-6.0-library folder
- Then in the “C:\dev\CGAL-6.0” directory I pasted that folder. 
- I have installed boost_1_86_0 in the “C:\boost_1_86_0” directory.

In the C++ directory inclusion files I have the following: 

`C:\dev\CGAL-6.0\include;C:\dev\CGAL-6.0\auxiliary\gmp\include;C:\boost_1_86_0;%(AdditionalIncludeDirectories)`

In the linking files: 
`C:\dev\CGAL-6.0\lib;C:\dev\CGAL-6.0\auxiliary\gmp\lib;C:\boost_1_86_0\libs;C:\dev\CGAL-6.0\lib\cmake\CGAL;%(AdditionalLibraryDirectories)`

I have added lib 2 times because I didn't know if I made a mistake so I added it just in case but it keeps giving the same error.

For the record, I read the documentation and tried to follow the installation steps but in the end that's what happened...

I tried to use vcpkg but it gets stuck when it starts downloading a gmp dependency I am using windows 11, you should check because that method didn't work for me. 

The code I am trying to execute is as follows:

```c++
#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
#include <CGAL/Projection_traits_xy_3.h>
#include <CGAL/Delaunay_triangulation_2.h>

#include <fstream>


typedef CGAL::Exact_predicates_inexact_constructions_kernel K;
typedef CGAL::Projection_traits_xy_3<K>  Gt;
typedef CGAL::Delaunay_triangulation_2<Gt> Delaunay;

typedef K::Point_3   Point;

int main()
{
	std::ifstream in("data/terrain.cin");
	std::istream_iterator<Point> begin(in);
	std::istream_iterator<Point> end;

	Delaunay dt(begin, end);
	std::cout << dt.number_of_vertices() << std::endl;
	return 0;
}
```

**Comment by sloriot** [comment]
You need to link with GMP. I guess you did not use cmake otherwise it would have been automatically added.

**Reply by DarkSun812** [reply]
How do I link gmp? I tried to use cmake but when I got to the make command it didn't recognize the installation files, that's where I'm having problems.

**Reply by sloriot** [reply]
Go into the properties of your project and in the linker section you can add the lib in input. Have a look at the doc of visual studio if you can't find it.

**Reply by DarkSun812** [reply]
If you look at the explanation, there I did that step of linking the library files in visual studio, but I still get the error. I don't know if I'm choosing the wrong paths...

**Reply by sloriot** [reply]
you only provided the path, you also need to provide the libs. 
What is the problem with cmake? did you pick visual studio as generator?

**Reply by DarkSun812** [reply]
Excuse my ignorance, but by including the path of C:\dev\CGAL-6.0\auxiliary\gmp\lib in the visual studio library linker, wouldn't it be adding the libraries?

The problem that have with cmake is the next...

```
C:\dev\CGAL-6.0>cd build

C:\dev\CGAL-6.0\build>cmake ..
-- Building for: Visual Studio 17 2022
-- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.22631.
-- The CXX compiler identification is MSVC 19.41.34120.0
-- The C compiler identification is MSVC 19.41.34120.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.41.34120/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.41.34120/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Build CGAL from release in directory CGAL-6.0
-- Packagenames: CGAL-6.0
== Setting paths (DONE) ==

== Generate version files ==
-- CGAL_MAJOR_VERSION=6
-- CGAL_MINOR_VERSION=0
-- CGAL_BUGFIX_VERSION=0
-- Visual Leak Detector (VLD) is not found.
-- Targeting Visual Studio 17 2022
-- Target build environment supports auto-linking
-- Using VC toolset 143.
-- Generator uses intermediate configuration directory: $(Configuration)
-- USING CMake version: 3.23.0
-- System: Windows
== Generate version files (DONE) ==

-- Multi-configuration CMake generator: cannot display flags
== Detect external libraries ==
== Detect external libraries (DONE) ==

== Generating build files ==
== Generating build files (DONE) ==

== Setting paths ==
-- Configuring done
-- Generating done
-- Build files have been written to: C:/dev/CGAL-6.0/build

C:\dev\CGAL-6.0\build>nmake

Microsoft Program Maintenance Utility (R) version 14.41.34120.0
(C) Microsoft Corporation. All rights reserved.Microsoft Program Maintenance Utility (R) version 14.41.34120.0
(C) Microsoft Corporation. All rights reserved.

NMAKE : fatal error U1064: no MAKE file found and no destination specified.
Stop.
```

And I don't know how to continue

**Reply by sloriot** [reply]
It should have generated a .sln file you can open with visual studio.

Adding a path does not a library. One directory might contains thousands of libraries.

**Reply by DarkSun812** [reply]
I tried to compile a sample project, but it does not recognize the QT6 installation, I will upload pictures of the results I get when I press the configure button.
![image](https://github.com/user-attachments/assets/37120120-a8a0-4fe3-8b78-23aba8c490ca)

I include the path to the bin folder but when I configure it says that the folder is not found.
I also added the environment variables that according to the documentation are necessary.
![image](https://github.com/user-attachments/assets/25e8462a-ebdf-476b-ac01-d34970e3e395)

Although I include the route, when I click on the configure button, it changes to the message that shows the first image.
![image](https://github.com/user-attachments/assets/f1db238f-8434-4dbe-97ab-90451a9425c2)

**Reply by DarkSun812** [reply]
In the folder I put in the path are these files, the ones it says it supposedly can't find

![image](https://github.com/user-attachments/assets/9f28903e-2f56-4c5e-9468-be4d43438582)

**Comment by ankofl** [comment]
I have the following task today: can I will give up using gmp-10.dll and mfpr-6.dll for moving on to Boost.Multiprecision? 

To the list of changes:

> GMP/MPFR are no longer mandatory to use CGAL, [Boost.Multiprecision](https://www.boost.org/doc/libs/1_72_0/libs/multiprecision/doc/html/index.html). can be used instead.

> The CGAL Core library is no longer based on GMP, but on [Boost.Multiprecision](https://www.boost.org/doc/libs/1_72_0/libs/multiprecision/doc/html/index.html). Either GMP backend or Boost backend can be used.

And how could I do that? 

P.S. In fact, the problem is that it is not very convenient for me to carry along with my dll and exe files gmp-10.dll and mpfr-6.dll I would like to get rid of this by using Boost.Multiprecision

**Reply by sloriot** [reply]
If you a recent enough boost version, simply do not provide gmp to cmake and boost multiprecision will be used by defaut. Otherwise, you can change the cmake variable `CGAL_CMAKE_EXACT_NT_BACKEND` to `BOOST_BACKEND` (by hand or using the drop-down menu in your cmake gui)

**Reply by ankofl** [reply]
Thank you very much!
